manifests/init.pp

```diff
@@ -348,5 +353,13 @@
       Class['jenkins::firewall'] ->
         Anchor['jenkins::end']
   }
+
+  if $config_template {
+    Anchor['jenkins::begin'] ->
+      Class[$jenkins_package_class] ->
+        Class['jenkins::config'] ->
+          Class['jenkins::config::global'] ->
```

Comment:
`config.xml` needs to be sync'd before jenkins service [re]starts so there needs to be ordering with respect to the the service class.